improver/utilities/compare.py

```diff
@@ -193,31 +188,28 @@ def compare_vars(name, actual_ds, desired_ds, rtol, atol,
     """
     if exclude_vars is None:
         exclude_vars = []
-    actual_vars = sorted(set(actual_ds.variables.keys()) - set(exclude_vars))
-    desired_vars = sorted(set(desired_ds.variables.keys()) - set(exclude_vars))
+    actual_vars = set(actual_ds.variables) - set(exclude_vars)
+    desired_vars = set(desired_ds.variables) - set(exclude_vars)
 
     if actual_vars != desired_vars:
-        msg = ("different variables - "
-               f"{name} {actual_vars} {desired_vars}")
-
+        msg = ("different variables - {name} "
```

Comment:
The ```{name}``` here is never set as the string is missing a leading ```f```.

```suggestion
        msg = (f"different variables - {name} "
```